pkgs/servers/nosql/cassandra/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv , fetchurl , jre , python , makeWrapper , gawk , bash , getopt , procps}:
+
+let
+
+  common = { version, sha256 }: stdenv.mkDerivation (rec {
+    name = "cassandra-${version}";
+
+    src = fetchurl {
+      url = "mirror://apache/cassandra/${version}/apache-${name}-bin.tar.gz";
+      inherit sha256;
+    };
+
+    nativeBuildInputs = [ makeWrapper ];
+
+    installPhase = ''
+      mkdir $out
+      mv * $out
+
+      for cmd in cassandra nodetool sstable*
+        do wrapProgram $out/bin/$cmd \
+          --set JAVA_HOME ${jre} \
+          --prefix PATH : ${bash}/bin \
+          --prefix PATH : ${getopt}/bin \
+          --prefix PATH : ${gawk}/bin \
+          --prefix PATH : ${procps}/bin
+      done
+
+      wrapProgram $out/bin/cqlsh --prefix PATH : ${python}/bin
+      '';
+
+    meta = with stdenv.lib; {
+      homepage = http://cassandra.apache.org/;
+      description = "A massively scalable open source NoSQL database";
+      platforms = platforms.all;
+      license = licenses.asl20;
+      maintainers = with maintainers; [ nckx rushmorem markus1189 ];
+    };
+  });
+
+in {
+
+ cassandra_2_1 = common {
+    version = "2.1.13";
+    sha256 = "09b3vf5jsv70xlfimj30v8l1zw7c5xdgpw5fpmn6jh8n3gigybqh";
+  };
```

Comment:
According to https://cassandra.apache.org/download/: 

> Apache Cassandra 2.1 is supported until November 2016 with critical fixes only. The latest release is 2.1.13, released on 2016-02-08.

But from my side we could also remove 2.1.13